### ISSUE TITLE
"entropy" has two meanings depending on the computational package

### ISSUE THREAD

**Issue by schneiderfelipe** [cclib/cclib#663]
Hi, it's me again.

Entropies extracted from Gaussian (09C01) or ORCA (4.0.1.2) logfiles have different meanings:

```
% # First a Gaussian logfile:
% ccget entropy enthalpy freeenergy gaussian_out.out
Attempting to read gaussian_out.out
entropy
0.00016293141036397934
enthalpy
-552.415496
freeenergy
-552.464074
```

Results from Gaussian assume $G = H - T S$ as usual (`entropy` means $S$ here):

```
In [1]: -298.15 * 0.00016293141036397934 + -552.415496                          
Out[1]: -552.464074
```

```
% # Now a ORCA logfile:
% ccget entropy enthalpy freeenergy orca_out.out
Attempting to read orca_out.out
entropy
0.04797662
enthalpy
-612.38317299
freeenergy
-612.43114961
```

But in results from ORCA, `entropy` means $T S$!

```
In [2]: -0.04797662 + -612.38317299                                             
Out[2]: -612.43114961
```

Is this wanted? I can imagine algorithms using ccclib not working across packages because of this.

**Comment by langner** [comment]
Hi Felipe, we do want consistency of attributes between parsers, of course. We recently had a similar discussion about dispersion correction to the SCF energy in this context (#664).

Which definition would you suggest making the standard one?

**Comment by schneiderfelipe** [comment]
Hi Karol,

Although I like how ORCA defines it, defining `entropy`as temperature times entropy can lead to misunderstandings. ORCA minimises this by explaining each definition in the logfile.

The way entropy is defined by Gaussian is much less error-prone. The user knows right away what `entropy` means, no need for explanations. I like this one better.

**Comment by langner** [comment]
OK, then let's go the Gaussian route, if possible.

@berquist @ATenderholt any doubts?

**Comment by berquist** [comment]
If we can be sure that every program will print the entropy without being multiplied by the temperature, then we should do that (the standard definition).

**Comment by langner** [comment]
> If we can be sure that every program will print the entropy without being multiplied by the temperature, then we should do that (the standard definition).

I dunno if we can really be sure for every program, but we can spot check the value (as we do in unit tests). Nonetheless, I don't want to go too far down this rabbit hole, because the purpose of cclib is not to validate the data programs spit out, just to parse it. It is reasonable, I think, to parse and add things together or whatnot in situations where that's clearly the right thing to do. But I don't want to lose sleep over ambiguous cases unless somebody brings it up and investigates what the right thing to do is. I'm fairly confident everyone shares this attitude :)

Given the above, it seems to me there are three clear action items:
* Document this in the docs!
* Gaussian prints the thing we want, but we don't seem to test it so add that bit.
* ORCA entropy needs to be divided by temperature (I assume it's printed in the logfile).
* QChem parses entropy, but we don't test, so add that bit. See if the value makes sense.
* Molcas parses it and there is a test, but the value is weird. Figure out if that's an issue related to the temperature, coordinates, or something else we don't care to dive into.

Is there anything else we want to do here at this point?

**Comment by Chokerino** [comment]
From what i understand this issue involves -
1) Changing `self.entropy = float(next(inputfile).split()[4])`  to `self.entropy` = float(next(inputfile).split()[4]/self.temperature)` in the orca parser.
2)Adding a comment is that file to show the same. If this needs to be odcument anywhere else please let me know.
3)The Qchem parser has a TODO comment `# 'entropy' (incorrect)`

**Comment by langner** [comment]
@Chokerino it would also be prudent to make sure tests enforce that difference.

**Comment by schneiderfelipe** [comment]
I would like to assign myself to this.

- [ ] Divide ORCA entropy by minus temperature (a sign change is needed)
- [ ] Test Gaussian and QChem in this respect, they seem to behave as we expect
- [ ] Document (especially units and sign)
- [ ] Check the value that is tested for Molcas
- [ ] Try to ensure the standard definition of `entropy` for other supported programs (to the extent that this is possible)

**Comment by schneiderfelipe** [comment]
> * Molcas parses it and there is a test, but the value is weird. Figure out if that's an issue related to the temperature, coordinates, or something else we don't care to dive into.

@langner The relevant part of the Molcas logfile is the following:

```
 *****************************************************
 Temperature =   298.15 Kelvin, Pressure =   1.00 atm
 -----------------------------------------------------
 Molecular Partition Function and Molar Entropy:
                        q/V (M**-3)    S(kcal/mol*K)
 Electronic            0.100000D+01        0.000
 Translational         0.453903D+29       40.502
 Rotational            0.613786D+06       29.465
 Vibrational           0.125395D-80       14.138
 TOTAL                 0.349349D-46       84.106
```

The value matches the value being tested (`0.13403144249776794 * 627.50947414 == 84.106`).
But, as the calculations for different temperatures *change the entropy*, I came to realize that this little asterisk in `S(kcal/mol*K)` might mean `(kcal/mol)*K` instead of `kcal/(mol*K)`.

So, Molcas seems to do the same thing as ORCA.

**Comment by langner** [comment]
Filipe, your diagnosis seems well founded - could you add a comment to the documentation, and perhaps also a short one around the code that parses this?

Also, could you elaborate what you mean by this, please:
> I thought Molcas would do the same as ORCA, but it reports non-zero entropies at 0 K.

**Comment by schneiderfelipe** [comment]
> Filipe, your diagnosis seems well founded - could you add a comment to the documentation, and perhaps also a short one around the code that parses this?

Sure, I'll do it as soon as I can.

> Also, could you elaborate what you mean by this, please:
> 
> > I thought Molcas would do the same as ORCA, but it reports non-zero entropies at 0 K.

ORCA gives entropy in units of energy because it pre multiplies "true entropy" by temperature (it gives T*S, so in #812, I divided ORCA's value by temperature). That means ORCA says zero at zero temperature.

Molcas does not do this, as it reports non-zero values at 0 K in `dvb_ir.out`. But, by working the entropy value out of the G - H difference, I found out that the reported value is off by a thousand. That means Molcas say `kcal/(mol*K)` but mean `cal/(mol*K)`.

By means of 33fcf661dd65c7c14329df981460108496b3a70e we can now be sure that all `entropy`, `enthalpy` and `freeenergy` values agree among each other by satisfying `G = H - TS` for all programs from which we parse those values.
